### PR TITLE
2019 November upgrade: Schnorr OP_CHECKMULTISIG spec

### PR DESCRIPTION
**Pull request by markblundeberg**

### CODE REVIEW

File: spec/2019-11-15-schnorrmultisig.md [hunk 1]
```diff
@@ -37,50 +37,51 @@ Mode 2 (new Schnorr support, M-of-N; consumes N+M+3 items from stack):
...
+    Set a cursor on the top signature (last signature pushed on stack).
+    Set another cursor on the top public key (last key pushed on stack).
     Calculate scriptCode.
     If activated, and the dummy element is not null, then:
         # New mode (2)
-        Convert the dummy element to a number, called checkbits.
-        If checkbits < 0, fail script.
+        Fail if the dummy element does not have length = floor((N+7)/8)
+        Set checkbits := 0, then iterate over the bytes in the dummy element in reverse order:
+            For each byte X,  checkbits := (checkbits << 8) | X
```
Comment (Mengerian) [inline spec/2019-11-15-schnorrmultisig.md]: This part seems confusing to me... What is "X"? a number?

File: spec/2019-11-15-schnorrmultisig.md [hunk 2]
```diff
@@ -100,19 +101,21 @@ The mechanics of CHECKMULTISIG are complicated due to the order of signature che
...
-- For M=0, the opcode always evaluates True. This also requires `checkbits=0` and can normally only occur in the legacy mode.
-- The new mode is actually designed to function as a full multisignature --- even including the `checkbits=0` case, though it will not be accessible in real transactions. Implementations can internally test this by disabling the minimal number encoding flag/rule, and by passing the length-1 byte array `{0x00}` for the dummy element; this is a non-null element that will activate the new mode, and yet count as `checkbits=0`.
+- A soft-failing CHECKMULTISIG (that returns False on stack) can only occur with all null signatures, due to NULLFAIL. For simplicity and avoiding malleability, the new mode does not allow a failing case, and a soft-failing CHEKCKMULTISIG must execute in the legacy mode. Note that even such a soft-failing checkmultisig still requires the top public key to be correctly encoded, due to the legacy mechanics.
```
Comment (Mengerian) [inline spec/2019-11-15-schnorrmultisig.md]: I would add that soft-failing CHECKMULTISIG must have dummy element = null ( even thought that's redundant info )

the second sentence could be: 
"For simplicity and avoiding malleability, the new mode does not allow a failing case, and a soft-failing CHECKMULTISIG must execute in the legacy mode (with null dummy element)."

Also CHEKCKMULTISIG -> CHECKMULTISIG

File: spec/2019-11-15-schnorrmultisig.md (hunk 1 above)
Comment (Mengerian) [reply spec/2019-11-15-schnorrmultisig.md]: On re-reading, it seems clear to me now... so maybe it is fine as-is

**Comment by markblundeberg** [comment]
@Mengerian I've rebased and updated per requests.

**Pull request merged**